src/radical/pilot/task.py

```diff
@@ -157,6 +157,18 @@ def _update(self, task_dict, reconnect=False):
         current = self.state
         target  = task_dict['state']
 
+        # we never update once the task is FAILED or DONE
+        if current in [rps.FAILED, rps.DONE]:
+            self._log.debug('task %s is final, ignore update', self.uid)
+            return
+
+        # when in CANCELED state, we only allow updates for `DONE` tasks - in
+        # that case the cancel command raced the task execution, and the
+        # execution actually won, so we don't want to waste that work
+        if current == rps.CANCELED and target != rps.DONE:
+            self._log.debug('task %s was CANCELED, ignore update', self.uid)
+            return
```

Comment:
should we still make an update? we need to ensure that if there are slots, we need to update them (probably some other attributes as well)

Reply:
Ack, done.